docs/design/proposals/decorators.md

```diff
@@ -0,0 +1,265 @@
+                      }
+                    }
+                  }
+                }).sort(p, p.path) }
+```
+
+## The Blast Radius Problem
+
+Decorators have broad blast radius. A small change to a decorator has the potential to impact every
+object in the cluster of a given kind. However, this problem is not new to decorators is is
+identical to the blast radius problem inherent to collections. We consider this to be an opportunity
+for a future KREP.
+
+## The Singleton Problem
+
+ResourceGraphDefinitions result in the creation of a new `kind` defined in the `schema`, which
+allows for the creation of multiple instances of the `kind`. We assert that this is an anti-pattern
+for most decorators, which should only ever have a single instance. If multiple instances of a
+decorator are defined they will take the same actions and cause write conflicts. We refer to this
+class of RGD as `Singletons`. Notably singletons don't require any spec or status in the `schema`.
+They only contain `apiVersion` and `kind` because those are required to instantiate the RGD. It's
+possible to define RGDs that use the decorator pattern AND contain schema (i.e., dynamic
+decorators), but use cases that demand this flexibility are unclear.
+
+The singleton pattern is useful beyond decorators. For example, software that uses the Kubernetes
+Operator Pattern (i.e. KRO, Karpenter, etc) is intended to be installed a **single** time in a
+cluster. These components contain cluster-scoped objects (e.g. CRDs) that can only be installed
+once. Customers looking to deploy these components using KRO would need to be careful to only
+instantiate the RGD once. Further, they must first create an RGD and then instantiate it without any
+schema. This is cumbersome and could be collapsed into a single API without losing functionality.
+
+Singletons are similar to how customers use helm today -- it's possible to deploy a chart multiple
+times, and customers self-enforce these singleton properties. For similar reasons, we don't consider
+this to be blocking for KREP-003 Decorators. We consider this to be an opportunity for a future
+KREP.
+
+Directionally, we posit that `singletons` could take the following form:
+
+1. The `schema` field is no longer required in RGDs
+2. An RGD without a `schema` would self-instantiate and create all of its resources as a singleton
+3. An RGD without a `schema` becomes the root node of the DAG, rather than the RGD's instance
+
+## The ExternalRef Naming Problem
+
+The name `ExternalRef` suggests a singular reference, which becomes misleading when referring to
+collections of resources. Additionally, the term `external` creates ambiguity about scope - it's
+unclear whether this refers to objects outside the RGD itself or objects outside the cluster
+entirely (as in multi-cluster scenarios). This awkwardness led me to explore alternative names.
+
+The name `watch` is a natural fit to both single resources and collections of resources. Watch is
```

Comment:
For me the term watch is not a "natural fit" - because we dont even watch them yet: https://github.com/kubernetes-sigs/kro/pull/611 would introduce this (we have a dedicated issue list for watching child resources, its not just about that PR), but not just for the current externalRefs, but also for other resources. So, watch is a misnomer because we actually would watch them all. Also currently the proposal is to have periodic mode instead of watches so maybe that would clash.

TLDR: to have "watch" be better than "externalRefs" we need to actually start watching them and make that a core principle. 

Currently externalRefs are mostly present to be included into the graph as references, thus the name.